python/pyspark/pandas/tests/test_dataframe.py

```diff
@@ -2910,7 +2920,12 @@ def get_data(left_columns=None, right_columns=None):
         left_pdf.update(right_pdf)
         left_psdf.update(right_psdf)
         self.assert_eq(left_pdf.sort_values(by=["A", "B"]), left_psdf.sort_values(by=["A", "B"]))
-        self.assert_eq(psser.sort_index(), pser.sort_index())
+        # SPARK-38946: Since Spark 3.4, df.update generate a new dataframe to follow
+        # pandas 1.4 behaviors
+        if LooseVersion(pd.__version__) >= LooseVersion("1.4"):
+            self.assert_eq(psser.sort_index(), pser.sort_index())
```

Comment:
ditto.

Reply:
See https://github.com/apache/spark/pull/36353#issuecomment-1178956945 , change the version to 1.4.0-1.4.2